src/components/core/geoChart/GeoChartInner.tsx

```diff
@@ -0,0 +1,373 @@
+import { isFluidLegendEnabled, isColorAssignmentItemChanged } from "../../../helpers/geoChart/common";
+import { getAvailableLegends } from "../../../helpers/geoChart/data";
+import { HEIGHT_OF_SIZE_LEGEND } from "./legends/PushpinCategoryLegend";
+import { IColorAssignment } from "../../../interfaces/Config";
+import { shouldShowFluid } from "../../../helpers/utils";
+
+const HEIGHT_OF_COLOR_LEGEND = 210;
+
+export function renderChart(props: IGeoChartRendererProps): React.ReactElement {
+    return <GeoChartRenderer {...props} />;
+}
+
+export function renderLegend(props: IGeoChartLegendRendererProps): React.ReactElement {
+    return <GeoChartLegendRenderer {...props} />;
+}
+
+export interface ICoreGeoChartProps extends ICommonChartProps, IDataSourceProviderInjectedProps {
+    config?: IGeoConfig;
+    documentObj?: Document;
+    chartRenderer?: (props: IGeoChartRendererProps) => React.ReactElement;
+    legendRenderer?: (props: IGeoChartLegendRendererProps) => React.ReactElement;
+    onCenterPositionChanged?: (center: IGeoLngLat) => void;
+    onZoomChanged?: (zoom: number) => void;
+    geoChartOptions?: IGeoChartInnerOptions;
+}
+
+export type IGeoChartInnerProps = ICoreGeoChartProps &
+    ILoadingInjectedProps &
+    IDataSourceProviderInjectedProps &
+    WrappedComponentProps;
+
+export interface IGeoChartInnerState {
+    enabledLegendItems: boolean[];
+    showFluidLegend: boolean;
+    colorAssignmentItem: IColorAssignment[];
+}
+export interface IGeoChartInnerOptions {
+    geoData: IGeoData;
+    categoryItems: IPushpinCategoryLegendItem[];
+    colorStrategy: IColorStrategy;
+    colorPalette: IColorPalette;
+}
+/**
+ * Geo Chart react component
+ */
+export class GeoChartInner extends React.PureComponent<IGeoChartInnerProps, IGeoChartInnerState> {
+    public static defaultProps: Partial<IGeoChartInnerProps> = {
+        ...commonDefaultProps,
+        documentObj: document,
+        chartRenderer: renderChart,
+        legendRenderer: renderLegend,
+        config: {
+            mapboxToken: "",
+        },
+    };
+
+    public static getDerivedStateFromProps(
+        nextProps: IGeoChartInnerProps,
+        prevState: IGeoChartInnerState,
+    ): Partial<IGeoChartInnerState> {
+        const {
+            geoChartOptions: { categoryItems, colorStrategy },
+        } = nextProps;
+        const colorAssignmentItem = colorStrategy.getColorAssignment();
+
+        if (!isColorAssignmentItemChanged(prevState.colorAssignmentItem, colorAssignmentItem)) {
+            return null;
+        }
+
+        return {
+            enabledLegendItems: new Array(categoryItems.length).fill(true),
+            colorAssignmentItem,
+        };
+    }
+
+    private readonly throttledOnWindowResize: ReturnType<typeof throttle>;
+    public constructor(props: IGeoChartInnerProps) {
+        super(props);
+        this.state = {
+            enabledLegendItems: [],
+            showFluidLegend: shouldShowFluid(this.props.documentObj),
```

Comment:
`this.props.documentObj` => `props.documentObj`

Reply:
fixed